prometheus/gc_collector.go

```diff
@@ -0,0 +1,43 @@
+package prometheus
+
+import (
+	"runtime/debug"
+)
+
+type gcCollector struct {
+	numGC      Gauge
+	pauseTotal Gauge
+}
+
+// NewGCCollector returns a collector which exports metrics about the
+// garbage collection stats of the current process.
+func NewGCCollector() *gcCollector {
+	return &gcCollector{
+		numGC: NewGauge(GaugeOpts{
+			Name: "process_gc_num_total",
+			Help: "Total number of garbage collector incovations for the current process.",
```

Comment:
Updated the code and moved things over to the `go_collector.go` file.

I'm fine with not using a summary but I'm wondering why to use counters instead of gauges? The golang runtime is doing the counting for us so using those absolute numbers a gauges makes sense to me. What am I missing?